### PR TITLE
Master new general setting abh

### PR DESCRIPTION
**Pull request by abh-odoo**
Description of the issue/feature this PR addresses:
Task: https://www.odoo.com/web?#id=29501&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720

Pad: https://pad.odoo.com/p/r.3d7984f6f3ed56f4652d535ebd6bdce0

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/auth_oauth/views/res_config_views.xml [hunk 1]
```diff
@@ -5,105 +5,33 @@
...
-            <xpath expr="//group[@name='google']/label" position="before">
-                <label for="id" string="Authentication" attrs="{'invisible':[('module_auth_oauth','=',False)]}"/>
-                <div attrs="{'invisible':[('module_auth_oauth','=',False)]}">
-                    <div name="google" id="top">
-                        <div>
-                            <field name="auth_oauth_google_enabled"/>
-                            <label for="auth_oauth_google_enabled" string="Allow users to sign in with Google"/>
+                <xpath expr="//div[@id='auth_oauth']" position="replace">
+                    <div class="mt16"/>
+                    <div>
```
Comment (atp-odoo) [inline addons/auth_oauth/views/res_config_views.xml]: why two div you can directly give class="mt16" here .

File: addons/auth_oauth/views/res_config_views.xml [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
-        <field name="name">base.config.settings.form.inherit</field>
-        <field name="model">base.config.settings</field>
-        <field name="inherit_id" ref="base_setup.view_general_configuration"/>
-        <field name="arch" type="xml">
-            <xpath expr="//div[@id='auth_oauth']" position="replace">
-                <div class="mt16">
-                    <button type="action" name="%(auth_oauth.action_oauth_provider)d" string="OAuth Providers" icon="fa-arrow-right" class="btn-link"/>
-                </div>
-            </xpath>
+        <record id="view_general_configuration" model="ir.ui.view">
```
Comment (amigrave) [inline addons/auth_oauth/views/res_config_views.xml]: Why this indent noise in the diff ?

File: addons/base_setup/views/res_config_views.xml [hunk 3]
```diff
@@ -9,147 +9,240 @@
...
-                                    When you send a document to a customer
-                                    (quotation, invoice), your customer will be
-                                    able to signup to get all his documents,
-                                    read your company news, check his projects,
-                                    etc.
-                                </p>
+                                <h2>Company</h2>
+                                <div class="row mt16 o_settings_container">
+                                    <div class="col-md-6 o_setting_box">
+                                        <div class="input-group">
```
Comment (atp-odoo) [inline addons/base_setup/views/res_config_views.xml]: remove this div. no input-group div anymore.

File: addons/base_setup/views/res_config_views.xml [hunk 4]
```diff
@@ -9,147 +9,240 @@
...
+                                            <div class="o_setting_left_pane"/>
+                                            <div class="o_setting_right_pane">
+                                                <label string="Company Details"/><i class="fa fa-building" aria-hidden="true"></i>
```
Comment (atp-odoo) [inline addons/base_setup/views/res_config_views.xml]: If user has multi companies access right, then only show this icon on next to box title at top right with following tooltip: Values set here are company-specific.

File: addons/base_setup/views/res_config_views.xml [hunk 5]
```diff
@@ -9,147 +9,240 @@
...
-                                <button type="action"
-                                    name="%(base.action_ir_mail_server_list)d"
-                                    string="Configure outgoing email servers" class="oe_link"/>
-                            </div>
-                        </div>
-                    </group>
-                    <group>
-                        <label for="id" string="Portal access"/>
-                        <div>
+                    <div class="o_form_sheet_bg">
```
Comment (atp-odoo) [inline addons/base_setup/views/res_config_views.xml]: o_form_sheet_bg ???

File: addons/base_setup/views/res_config_views.xml [hunk 6]
```diff
@@ -9,147 +9,240 @@
...
+                                                                <field name="country_code" class="col-md-4 text-muted"/>
+                                                                <field name="company_zip" class="col-md-4 text-muted"/>
+                                                                <field name="company_country" class="col-md-12 text-muted"/>
+                                                            </div>
+                                                            <div class="col-md-4">
+                                                                <field name="company_logo" widget="image" class="oe_inline pull_left oe_avatar"/> 
+                                                            </div>
+                                                        </div>
+                                                    </div>
+                                                    <button type="object" name="open_company" string="Company Details" icon="fa-arrow-right" class="btn-link"/>
```
Comment (atp-odoo) [inline addons/base_setup/views/res_config_views.xml]: put this button inside div to make proper alignment..

File: addons/base_setup/views/res_config_views.xml [hunk 7]
```diff
@@ -9,147 +9,240 @@
...
+                                                </div>
+                                            </div>
+                                        </div>
+                                    </div>
+                                    <div class="col-md-6 o_setting_box">
+                                        <div class="input-group">
```
Comment (atp-odoo) [inline addons/base_setup/views/res_config_views.xml]: remove this div input-group everywhere .

File: addons/base_setup/views/res_config_views.xml [hunk 8]
```diff
@@ -9,147 +9,240 @@
...
+                                                </div>
+                                            </div>
+                                        </div>
+                                    </div>
+                                    <div class="col-md-6 o_setting_box" id="google_oauth">
+                                    </div>
                                 </div>
                             </div>
+                            <div name="company_dependent_settings_message">
+                                <div>(*) This configuration is related to the company you're logged into.</div>
```
Comment (atp-odoo) [inline addons/base_setup/views/res_config_views.xml]: is it required now? i don't think so. now we have set icon that will be visible on multi-company..

File: addons/fetchmail/views/fetchmail_views.xml [hunk 9]
```diff
@@ -96,19 +96,23 @@
...
+                    <div>
+                        <button type="action"
+                        name="%(base.action_ir_mail_server_list)d"
+                        string="Outgoing Email Servers" icon="fa-arrow-right" class="btn-link"/>
+                    </div>
+                </xpath>
             </field>
         </record>
 
-
```
Comment (atp-odoo) [inline addons/fetchmail/views/fetchmail_views.xml]: remove this unused diff

File: addons/google_drive/views/res_config_views.xml [hunk 10]
```diff
@@ -6,17 +6,18 @@
...
+                <div class="mt16 row">
+                    <label for="google_drive_authorization_code" class="col-md-3 oe_light_label"/>
+                    <field name="google_drive_authorization_code"/>
                 </div>
+                <div>
+                    <a type="button" class="oe-link" target="_blank"><i class="fa fa-fw fa-arrow-right"/>
+                    <field name="google_drive_uri" widget="url" text=" Generate Code"/> 
+                    </a>
+                </div>
+                <button type="action" name="%(action_google_drive_users_config)d" string="Google Drive Templates" icon="fa-arrow-right"  class="btn-link" />
```
Comment (atp-odoo) [inline addons/google_drive/views/res_config_views.xml]: inside div to remove alignment issue

File: addons/report/views/base_config_settings_views.xml [hunk 11]
```diff
@@ -5,19 +5,64 @@
...
+                            <field name="custom_footer"/>
+                        </div>
+                        <div class="o_setting_right_pane">
+                            <label string="Custom Footer" for="custom_footer"/>
+                            <div class="text-muted">
+                                Set a custom footer for your printed documents
+                            </div>
+                            <div class="content-group" attrs="{'invisible': [('custom_footer','=',False)]}">
+                                <div class="mt16">
+                                <field name="rml_footer"/>
```
Comment (atp-odoo) [inline addons/report/views/base_config_settings_views.xml]: improve indentation!!

**Pull request merged**